src/software/amazon/ion/impl/lite/IonIntLite.java

```diff
@@ -39,11 +39,17 @@
         IonType.INT.toString().hashCode();
 
 
-    private static final byte INT_SIZE = (byte)IntegerSize.INT.ordinal();
-    private static final byte LONG_SIZE = (byte)IntegerSize.LONG.ordinal();
-    private static final byte BIG_INTEGER_SIZE = (byte)IntegerSize.BIG_INTEGER.ordinal();
-    private static final byte NULL_SIZE = (byte)IntegerSize.values().length;
-    private static final IntegerSize[] SIZES = new IntegerSize[NULL_SIZE + 1];
+    // This mask combines the IS_BOOL_TRUE (0x08) and IS_IVM (0x10)
+    // masks from IonValueLite. Those flags are never relevant for
+    // IonInts, which makes them safe for reuse here.
+    private static final int INT_SIZE_MASK  = 0x18;
+    private static final int INT_SIZE_SHIFT = 0x03;
+
+    private static final int NULL_SIZE          = 0;
+    private static final int INT_SIZE           = 1;
+    private static final int LONG_SIZE          = 2;
+    private static final int BIG_INTEGER_SIZE   = 3;
+    private static final IntegerSize[] SIZES = new IntegerSize[4];
```

Comment:
It's `IntegerSize.values()` plus a special case for `null`. But the special case for `null` isn't even needed -- there's already a flag for a null value. So `SIZES` just becomes `IntegerSize.values()`, with an `isNullValue()` check in `getIntegerSize()`.